### PR TITLE
feat: support Istio add-on in AKS

### LINKED ISSUE #3363: [AVM Module Issue]: Support Istio Service Mesh add-on in AKS
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/container-service/managed-cluster

### (Optional) Module Version

_No response_

### Description

Hi,

Is it possible to support Istio Service Mesh add-on in AKS Module?

https://learn.microsoft.com/en-us/azure/aks/istio-about

Thanks!

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by fsforza**
## Description

This Pull Request enable Istio add-on on AKS. Following features are enabled:

- Istio Revisions Selection
- Internal Ingress Gateway
- External Ingress Gateway
- Plug your own CA from Key Vault

Closes #3363 

## Pipeline Reference

| Pipeline |
| -------- |
|      [avm.res.container-service.managed-cluster](https://github.com/fsforza/bicep-registry-modules/actions/workflows/avm.res.container-service.managed-cluster.yml)    |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by fsforza** [comment]
@microsoft-github-policy-service agree

File: avm/utilities/e2e-template-assets/scripts/Set-CertificateAuthorityInKeyVault.ps1
Comment (eriqua) [inline avm/utilities/e2e-template-assets/scripts/Set-CertificateAuthorityInKeyVault.ps1:1]: Hey @fsforza, the e2e-teplate-assets folder hosts general templates leveraged by multiple modules. As such, changes to this folder trigger deployment pipelines to all modules.

If this file is only used by the AKS module I'd suggest to move it to its e2e tests folder.

Comment (fsforza) [reply avm/utilities/e2e-template-assets/scripts/Set-CertificateAuthorityInKeyVault.ps1:1]: Hi @eriqua, I had made the script generic to make it reusable in other test scenarios of other bicep modules as well.

But no problem, I moved it

File: package-lock.json
Comment (eriqua) [inline package-lock.json:1]: Hi @fsforza, could you please elaborate on these changes and the ones in package.json? Are they intended?

Comment (fsforza) [reply package-lock.json:1]: Uh, I don't think so.
I think they were automatically changed by some VSCode extension.

I restored them from main branch

Comment (eriqua) [reply package-lock.json:1]: Thanks!

**Comment by eriqua** [comment]
Thanks a lot @fsforza for quickly addressing the above comments!
The scope of the change is now at module level, therefore the module owner can take the review further.
@ilhaan @JPEasier FYI

**Comment by eriqua** [comment]
Hey @ilhaan, @JPEasier just a heads up to have a look at this PR when you have some availability. Thanks!

**Review by JPEasier — APPROVED** [review]
Looks good for me!
Thank you for your contribution @fsforza 🚀

**Pull request merged**